### PR TITLE
update doc about kubesphere network policy

### PR DESCRIPTION
**Pull request by duanjiong**
Signed-off-by: Duan Jiong <djduanjiong@gmail.com>

### CODE REVIEW

File: sig-network/concepts-and-designs/kubesphere-network-policy.md [hunk 1]
```diff
@@ -48,23 +45,47 @@ spec:
...
-            matchLabels:
-              "kubesphere.io/workspace": ${namespace-name}
+  policyTypes:
+  - ingress
+```
+
+Let's say I have two projects, A and B. They are in same workspace. If the A and B have not set annotation "kubesphere.io/network-isolate = enabled", they can access each other. The same thing goes for services. If project A set annotation "kubesphere.io/network-isolate = enabled", A cannot access services in B, unless A allows egress for services in B.
+
+## The Namespace NetworkPolicy resource 
+See the [Namespace NetworkPolicy](https://github.com/kubesphere/kubesphere/blob/master/pkg/apis/network/v1alpha1/namespacenetworkpolicy_types.go#L31) reference for a full definition of the resource.
```
Comment (zheng1) [inline sig-network/concepts-and-designs/kubesphere-network-policy.md:55]: This link with the hashtag will be better `https://github.com/kubesphere/kubesphere/blob/d3bdcd0/pkg/apis/network/v1alpha1/namespacenetworkpolicy_types.go#L31`

File: sig-network/concepts-and-designs/kubesphere-network-policy.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ Now kubernetes defines NetworkPolicy CRD, we can use it to achieve network isola
 There two kinds of isolation i.e., workspace and project. Workspace uses a new field "networkIsolation" in spec, and project uses namespace annotation "kubesphere.io/network-isolate = enabled". The two types of isolation, which are different from the design of K8s network policy, cannot coexist, so we can just
```
Comment (FeynmanZhou) [inline sig-network/concepts-and-designs/kubesphere-network-policy.md:11]: There are

Comment (FeynmanZhou) [inline sig-network/concepts-and-designs/kubesphere-network-policy.md:11]: The two types of isolation, which are different from the design of K8s network policy, cannot coexist, so we can just
=> The two types of isolation are different from the design of K8s network policy, they cannot coexist in a cluster, so we can just choose one of them to achieve the isolation.

File: sig-network/concepts-and-designs/kubesphere-network-policy.md [hunk 3]
```diff
@@ -26,14 +26,11 @@ spec:
...
-    - To:
-        - namespaceSelector:
-            matchLabels:
-              "kubesphere.io/workspace": ${workspace-name}
+  policyTypes:
+  - ingress
 ```
 
-* When a project is isolated, the following Kubernetes NetworkPolicy will be added in the namespace, deny all incoming and outgoing traffic exception traffic in the same namespace:
+* When a project is isolated, the following Kubernetes NetworkPolicy will be added in the namespace, deny all incoming traffic exception traffic in the same namespace:
```
Comment (FeynmanZhou) [inline sig-network/concepts-and-designs/kubesphere-network-policy.md:33]: exception =》 except

File: sig-network/concepts-and-designs/kubesphere-network-policy.md [hunk 4]
```diff
@@ -48,23 +45,47 @@ spec:
...
-  Egress:
-    - To:
-        - namespaceSelector:
-            matchLabels:
-              "kubesphere.io/workspace": ${namespace-name}
+  policyTypes:
+  - ingress
+```
+
+Let's say I have two projects, A and B. They are in same workspace. If the A and B have not set annotation "kubesphere.io/network-isolate = enabled", they can access each other. The same thing goes for services. If project A set annotation "kubesphere.io/network-isolate = enabled", A cannot access services in B, unless A allows egress for services in B.
```
Comment (FeynmanZhou) [inline sig-network/concepts-and-designs/kubesphere-network-policy.md:52]: They are in the same workspace.

Comment (FeynmanZhou) [inline sig-network/concepts-and-designs/kubesphere-network-policy.md:52]: If the A and B have not been set an annotation "kubesphere.io/network-isolate = enabled", they can access each other. The same thing goes for services. If project A has been set an annotation "kubesphere.io/network-isolate = enabled", A cannot access services in B, unless A allows egress for services in B.

File: sig-network/concepts-and-designs/kubesphere-network-policy.md [hunk 5]
```diff
@@ -48,23 +45,47 @@ spec:
...
+
+## The Namespace NetworkPolicy resource 
+See the [Namespace NetworkPolicy](https://github.com/kubesphere/kubesphere/blob/master/pkg/apis/network/v1alpha1/namespacenetworkpolicy_types.go#L31) reference for a full definition of the resource.
+With the exception of this field `service`, the fields are the same as kubernetes NetworkPolicy.
+
+An example Namespace NetworkPolicy might look like this:
```
Comment (FeynmanZhou) [inline sig-network/concepts-and-designs/kubesphere-network-policy.md:58]: Let's take an example of NetworkPolicy for a Namespace:

File: sig-network/concepts-and-designs/kubesphere-network-policy.md [hunk 6]
```diff
@@ -48,23 +45,47 @@ spec:
...
+spec:
+  egress:
+    - to:
+      - service:
+          name: my-svc
+          namespace: my-project
 ```
 
-Beyond that, we also allow the bellowing traffic:
+The example policy contains a single rule, which matches traffic selected by a service.
```
Comment (FeynmanZhou) [inline sig-network/concepts-and-designs/kubesphere-network-policy.md:73]: The example policy contains a single rule, which matches the traffic selected by a service.

File: sig-network/concepts-and-designs/kubesphere-network-policy.md [hunk 7]
```diff
@@ -48,23 +45,47 @@ spec:
...
 
+When a policy which contain an egress rule to a service added, we also allow the bellowing egress traffic automatically:
```
Comment (FeynmanZhou) [inline sig-network/concepts-and-designs/kubesphere-network-policy.md:75]: When a policy contains an egress rule to service is added, the following egress traffic will be allowed automatically:

File: sig-network/concepts-and-designs/kubesphere-network-policy.md [hunk 8]
```diff
@@ -48,23 +45,47 @@ spec:
...
 * DNS traffic, only port 53 and protocol TCP/UDP
-* All nodes internal IP, such as IP on eth0
 
-Let's say I have two projects, A and B. They are in same workspace. If the A and B have not set annotation "kubesphere.io/network-isolate = enabled", they can access each other. The same thing goes for services. If project A set annotation "kubesphere.io/network-isolate = enabled", A cannot access services in B, unless A allows egress for services in B.
+## Source IP
 
-## Future
+When we use CIDR in Namespace NetworkPolicy, we should know what happens to the source IP of packets.
+
+Packets sent to ClusterIP/PodIP from within the cluster are never source NAT'd by default.
```
Comment (FeynmanZhou) [inline sig-network/concepts-and-designs/kubesphere-network-policy.md:82]: NAT'd?

File: sig-network/concepts-and-designs/kubesphere-network-policy.md (hunk 4 above)
Comment (FeynmanZhou) [reply sig-network/concepts-and-designs/kubesphere-network-policy.md:52]: use the shadow for words like `"kubesphere.io/network-isolate = enabled"`

File: sig-network/concepts-and-designs/kubesphere-network-policy.md [hunk 9]
```diff
@@ -48,23 +45,47 @@ spec:
...
+However, if you're setting masqueradeAll to true, the source IP will be changed to  node IP or overlay tunnel IP.
 
-Now it just works correctly when proxy mode is iptables, because ipvs will use src ip that is assigned by CNI plugin. But we cannot obtain this ip. we will support ipvs in future.
+Packets sent to Services with Type=LoadBalancer or NodePort are source NAT'd by default, because all schedulable Kubernetes nodes in the Ready state are eligible for load-balanced traffic. So if packets arrive at a node without an endpoint, the system proxies it to a node with an endpoint, replacing the source IP on the packet with the IP of the node.
+However, if you're setting the service.spec.externalTrafficPolicy field to Local, the packets will only be sent to the node which they arrived, and the source IP will not be replaced.
```
Comment (FeynmanZhou) [inline sig-network/concepts-and-designs/kubesphere-network-policy.md:86]: service.spec.externalTrafficPolicy => `service.spec.externalTrafficPolicy`

Comment (FeynmanZhou) [reply sig-network/concepts-and-designs/kubesphere-network-policy.md:86]: which they arrived => where they arrived

File: sig-network/concepts-and-designs/kubesphere-network-policy.md (hunk 8 above)
Comment (duanjiong) [reply sig-network/concepts-and-designs/kubesphere-network-policy.md:82]: I use it same as kubenetes.io
https://kubernetes.io/docs/tutorials/services/source-ip/

**Pull request merged**